src/shared/hooks/useRiseAnimation.ts

```
@@ -0,0 +1,37 @@
+// 채팅 버블에서 "상대편이 입력을 시작하는 순간" 반대쪽 버블을 그 입력 영역 쪽에서
+// 올라오듯 보여주는 데 쓴다. getRiseDistance를 넘기면 트리거되는 그 순간에 실제 두 영역
+// 사이의 거리를 재서 시작 위치(translateY)로 사용하고, 없으면 CSS 기본값을 쓴다.
+export const useRiseAnimation = (
+  trigger: boolean,
+  getRiseDistance?: () => number,
+) => {
+  const [isRising, setIsRising] = useState(false);
+  const [riseDistance, setRiseDistance] = useState(0);
+  const prevTriggerRef = useRef(trigger);
+  const getRiseDistanceRef = useRef(getRiseDistance);
+  getRiseDistanceRef.current = getRiseDistance;
```

ESLint

[error] Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

/home/jailuser/git/src/shared/hooks/useRiseAnimation.ts:18:3
  16 |   const prevTriggerRef = useRef(trigger);
  17 |   const getRiseDistanceRef = useRef(getRiseDistance);
> 18 |   getRiseDistanceRef.current = getRiseDistance;
     |   ^^^^^^^^^^^^^^^^^^^^^^^^^^ Cannot update ref during render
  19 |
  20 |   useEffect(() => {
  21 |     const justStartedTyping = trigger && !prevTriggerRef.current;

(react-hooks/refs)